### PR TITLE
Feat/session title generation

### PR DESCRIPTION
**Pull request by riturajFi**
## Why we need this PR?
This PR introduces automatic session title generation from the first meaningful user message so session lists are easier to scan and manage.

Today, sessions can remain untitled/opaque in UI/API consumers, especially when many sessions exist. This change adds a nullable `display_title` field and populates it once, non-blockingly, during pending-message processing.

> This should close Issue (#118)

### What changed
1. Added `display_title` to `Session` models in both services (API + CORE ORM sync):
- `src/server/api/go/internal/modules/model/session.go`
- `src/server/core/acontext_core/schema/orm/session.py`

2. Added data-layer helpers in CORE:
- `update_session_display_title(...)`: persists generated title
- `should_generate_session_display_title(...)`: one-time gate (skip if already set)
- Renamed ORM entity variable usage to `session_record` for clarity
- File: `src/server/core/acontext_core/service/data/session.py`

3. Implemented title generation flow in message controller:
- Extract first user text from pending session messages
- Normalize and quality-gate input (`min length`, skip low-signal inputs like `hi`, `test`, etc.)
- Generate title via `llm_complete(...)` with dedicated prompt id: `session.display_title.first_user`
- Sanitize output (strip quotes/newlines, trim length, require alphanumeric content)
- Fallback to first-user-message words if model output is unusable
- Persist title only when valid
- Entire flow is non-blocking: all failures are caught and logged; task extraction continues
- Title generation runs **before** `task_agent_curd` (description aligned with code)
- Renamed DB context vars to `db_session` to avoid shadowing
- File: `src/server/core/acontext_core/service/controller/message.py`

4. Added CORE runtime schema patch organization for existing deployments:
- Extracted runtime schema patch logic into:
  - `src/server/core/acontext_core/infra/schema_migrations.py`
- Wired `db.py` to call patch application after `create_all`:
  - `src/server/core/acontext_core/infra/db.py`
- Patch includes:
  - `ALTER TABLE sessions ADD COLUMN IF NOT EXISTS display_title TEXT;`

5. Updated SDK types for session response shape:
- Python SDK `Session` type includes nullable `display_title`:
  - `src/client/acontext-py/src/acontext/types/session.py`
- TypeScript SDK `SessionSchema` includes nullable `display_title`:
  - `src/client/acontext-ts/src/types/session.ts`
- Added parsing tests:
  - `src/client/acontext-py/tests/test_client.py`
  - `src/client/acontext-py/tests/test_async_client.py`
  - `src/client/acontext-ts/tests/mocks.ts`
  - `src/client/acontext-ts/tests/client.test.ts`

### Behavioral guarantees
- Title generation is best-effort and never blocks main message processing.
- Title is generated once per session unless empty/missing.
- Output is constrained to short, readable titles.

### Non-functional notes
- Removed local/editor artifact from PR scope:
  - `.vscode/settings.json` (removed)
- `src/server/.env.local-api` is not part of this PR change set.

## Implementation Tasks
Please ensure your pull request meets the following requirements:
- [x] Add nullable `display_title` to API session model (`src/server/api/go/internal/modules/model/session.go`)
- [x] Add nullable `display_title` to CORE ORM session model (`src/server/core/acontext_core/schema/orm/session.py`)
- [x] Add CORE data helpers to gate and persist display titles (`src/server/core/acontext_core/service/data/session.py`)
- [x] Add first-user-message extraction, quality checks, LLM title generation, sanitization, and persistence in message controller (`src/server/core/acontext_core/service/controller/message.py`)
- [x] Ensure title generation path is non-blocking with defensive error handling and logging
- [x] Organize runtime schema patch query logic in dedicated infra migration helper (`src/server/core/acontext_core/infra/schema_migrations.py`)
- [x] Update SDK session types for `display_title` (Python + TypeScript)
- [x] Remove `.vscode/settings.json` from PR scope

## Impact Areas
Which part of Acontext would this feature affect?
- [x] Client SDK (Python)
- [x] Client SDK (TypeScript)
- [x] Core Service
- [x] API Server
- [ ] Dashboard
- [ ] CLI Tool
- [ ] Documentation
- [ ] Other

## Checklist
- [x] Open your pull request against the `dev` branch.
- [ ] All tests pass in available continuous integration systems (e.g., GitHub Actions).
- [x] Tests are added or modified as needed to cover code changes.

### CODE REVIEW

File: src/server/core/acontext_core/infra/schema_migrations.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+from sqlalchemy import text
+from sqlalchemy.ext.asyncio import AsyncSession
+
+DISPLAY_TITLE_COLUMN_PATCH_NAME = "sessions.display_title"
+DISPLAY_TITLE_COLUMN_PATCH_SQL = text(
+    "ALTER TABLE sessions ADD COLUMN IF NOT EXISTS display_title TEXT;"
+)
+
+
+async def apply_runtime_schema_patches(db_session: AsyncSession) -> list[str]:
```
Comment (GenerQAQ) [inline src/server/core/acontext_core/infra/schema_migrations.py]: This runtime schema-patch approach (raw SQL at startup) bypasses the project's migration management. It can't be tracked, versioned, or rolled back, and may conflict with future migrations that touch the same table.

Please use a proper database migration (e.g., Alembic) to add the `display_title` column. This file and the `_apply_schema_migrations` call in `db.py` should be removed in favor of a migration script.

File: src/server/core/acontext_core/service/controller/message.py [hunk 2]
```diff
@@ -6,10 +6,125 @@
...
+from ...llm.complete import llm_complete
 from ...env import LOG
 from ...schema.config import ProjectConfig
 from ...telemetry.get_metrics import get_metrics
 from ...constants import ExcessMetricTags
+from ..data import session as SD
+
+TITLE_INPUT_MAX_CHARS = 512
+TITLE_INPUT_MIN_CHARS = 12
+TITLE_GENERATION_MAX_TOKENS = 24
```
Comment (GenerQAQ) [inline src/server/core/acontext_core/service/controller/message.py]: `24` max tokens works for English titles (3-8 words), but CJK and other non-English languages often use 2-3 tokens per character. A Chinese title of 8 characters could easily require 16-24 tokens, leaving no margin.

Consider increasing to **48-64** to support multilingual title generation.

File: src/server/core/acontext_core/service/controller/message.py [hunk 3]
```diff
@@ -6,10 +6,125 @@
...
 from ...llm.agent import task as AT
+from ...llm.complete import llm_complete
 from ...env import LOG
 from ...schema.config import ProjectConfig
 from ...telemetry.get_metrics import get_metrics
 from ...constants import ExcessMetricTags
+from ..data import session as SD
+
+TITLE_INPUT_MAX_CHARS = 512
+TITLE_INPUT_MIN_CHARS = 12
```
Comment (GenerQAQ) [inline src/server/core/acontext_core/service/controller/message.py]: 12 characters may be too strict as a minimum. Reasonable first messages like `"Fix this bug"` (12 chars) are borderline, and shorter but meaningful inputs like `"Debug my app"` (12 chars) barely pass.

Consider lowering to **8-10** characters.

File: src/server/core/acontext_core/service/controller/message.py [hunk 4]
```diff
@@ -6,10 +6,125 @@
...
+        return None
+    if len(normalized) > max_chars:
+        normalized = normalized[:max_chars].rstrip()
+    return normalized
+
+
+def check_title_input_quality(text: str | None) -> tuple[bool, str]:
+    if text is None:
+        return False, "empty"
+    normalized = normalize_title_input_text(text)
```
Comment (GenerQAQ) [inline src/server/core/acontext_core/service/controller/message.py]: Redundant normalization: `extract_first_user_message_text` already calls `normalize_title_input_text` on the extracted text before returning it (line 75). Calling `normalize_title_input_text` again here is a no-op on already-normalized input.

Consider removing this call and operating directly on `text`, or document why the double normalization is intentional.

File: src/server/core/acontext_core/service/controller/message.py [hunk 5]
```diff
@@ -65,14 +180,88 @@ async def process_session_pending_message(
...
+                        if eil:
+                            raise ValueError(eil.errmsg)
+                    LOG.debug(f"Persisted display_title for session {session_id}")
+            except Exception as title_err:
+                LOG.warning(
+                    f"Skip title generation/persist for session {session_id}: {title_err}"
+                )
+
+        ls_session = None
         async with DB_CLIENT.get_session_context() as session:
```
Comment (GenerQAQ) [inline src/server/core/acontext_core/service/controller/message.py:97]: This `session` variable was not renamed to `db_session`, inconsistent with the rest of the function. The PR description says all DB context variables were renamed to avoid shadowing.

```suggestion
        async with DB_CLIENT.get_session_context() as db_session:
```

**Review by GenerQAQ — CHANGES_REQUESTED** [review]
Thanks for the PR! The overall approach — non-blocking, best-effort session title generation with a one-time gate — is solid. There are several items to address before this can be merged.

---

### Must Fix

**1. Rebase on latest `dev`**
The branch is 1 commit behind `dev` — missing `c953a28 feat(session): add copy session functionality with SDK support (#280)`. Since that commit also touches session-related code and SDKs, please rebase to ensure compatibility.

**2. Too many plan files**
This PR includes **22 plan markdown files** (one per milestone/sub-task). These are intermediate development artifacts and shouldn't be committed to the repository — they add significant noise to `plans/`. Please remove all milestone-level plans and keep only one summary plan (e.g., `ai-session-display-title-first-message.md`).

**3. Replace runtime schema patch with proper migration**
`schema_migrations.py` applies raw SQL (`ALTER TABLE ... ADD COLUMN IF NOT EXISTS`) at application startup. This bypasses the project's migration management — it can't be tracked, versioned, or rolled back. Please use the standard database migration approach instead. See inline comment.

**4. Add CORE unit tests**
The pure helper functions (`normalize_title_input_text`, `check_title_input_quality`, `extract_first_user_message_text`, `sanitize_generated_title`) have **zero** test coverage. These are ideal candidates for unit testing — they are pure functions with well-defined inputs and outputs. Please add tests for edge cases (empty input, Unicode, CJK text, very long input, boundary lengths, etc.).

**5. Incomplete variable renaming (`session` → `db_session`)**
The PR description says all DB context variables were renamed from `session` to `db_session` to avoid shadowing, but the learning_space query block still uses `session`. See inline comment at line 258.

---

### Suggestions

See inline comments for:
- `TITLE_GENERATION_MAX_TOKENS` may be too small for non-English languages
- `TITLE_INPUT_MIN_CHARS` may be too strict
- Redundant `normalize_title_input_text` call in `check_title_input_quality`
